### DISCUSSION TITLE
Rich code navigation

### DISCUSSION THREAD

**Discussion by fredericDelaporte** [nhibernate/nhibernate-core#3237] [Ideas]
Shouldn't we enable rich code navigation on NHibernate ?

https://github.com/marketplace/actions/rich-code-nav-indexer
https://visualstudio.microsoft.com/services/rich-code-navigation/

**Comment by alaingaudreau** [comment]
From my understanding, the use is rather limited at the moment. 

afaik, it's only good for "find references" and a few other use cases.

The external packages are generally treated as black boxes with only the exposed points being of interest unless you are doing diagnostics (like we had to do for my year old unanswered NHibernate question) where you will need much more than just find references and to read code. 

It could be usefull for small packages but for say NHibernate that has so many tiny methods that cascade for miles on every event that is fired, you could never get a good picture of what it's doing and how from following a reference to one method or interface.

If they ever enable breakpoint mappings or real time tracing, that would make it interesting for tracing issues in external packages without having to use a local build with source code in order to properly trace back calls and data manipulation but from what I have read a while ago, that is years out if at all.
